snobal_1d/station_pipeline.py

```diff
@@ -0,0 +1,257 @@
+    :param RH_col: col with RH
+    :param sw_net: col with net shortwave, in W/M^2
+    :param air_t_col_K: col with 
+    :return: list populated with estimated longwave.
+    """
+    SB_CONST = 5.67e-8
+    
+    clear = {'c1': -75.28,
+             'c2': 0.82,
+             'c3': 0.79}
+    
+    cloudy = {'c1': -212.59,
+              'c2': 1.89,
+              'c3': 1.06}
+    
+    lw_in_est = []
+
+    # night branch
+    if sw_net < 10.0:
+        if RH_col < 80:
+            # clear branch
+            x = clear['c1'] + clear['c2'] * RH_col + clear['c3'] * SB_CONST * air_t_col_K ** 4
+            lw_in_est.append(x)
+        elif RH_col >= 80:
+            # cloudy branch
+            x = cloudy['c1'] + cloudy['c2'] * RH_col + cloudy['c3'] * SB_CONST * air_t_col_K ** 4
+            lw_in_est.append(x)
+    # day branch
+    elif sw_net >= 10:   
+        if RH_col < 60:
+            # clear branch
+            x = clear['c1'] + clear['c2'] * RH_col + clear['c3'] * SB_CONST * air_t_col_K ** 4
+            lw_in_est.append(x)
+        elif RH_col >= 60:
+            # cloudy branch
+            x = cloudy['c1'] + cloudy['c2'] * RH_col + cloudy['c3'] * SB_CONST * air_t_col_K ** 4
+            lw_in_est.append(x)
+    
+    return lw_in_est
+
+
+def estimate_lw(df, lw_col, temp_col, threshold):
+    """
+    Simplified (collapsed) longwave parameterization from Skiles et al. 2012. 
+    
+    :param df: dataframe with input data
+    :param lw_col: column with incoming longwave radiation
+    :param threshold: theshold incoming longwave (in w/m^2) below which the correction is applied. We assume snow is occluding
+                      the sensor when it is below this theshold.
+    :param precip_col: column with binned precipitation data. 
+    :return: estimate of lw_in during precip, based on air t, in new col "incoming_radiation_lw_corr".
+    """
+    # create new col for corrected data
+    df['incoming_radiation_lw_corr'] = np.nan
+    # convert air temp from C to K
+    df['air_temp_K'] = df[temp_col].apply(lambda x: x + 273.15)
+    df.loc[(df[lw_col] < threshold) & (df['fraction'] != 0), 'incoming_radiation_lw_corr'] = 0.963*SB_CONST*df['air_temp_K']**4
+    return df
+
+    
+def replace_sw_in(df, default_sw_in, alt_sw_in, precip_col):
+    """replace sw_in at Atwater with sw_in from Reynolds peak when Atwater radiometer is occluded."""
+    df.loc[(df[default_sw_in] < df[alt_sw_in]) & (df[precip_col] != 0), 'corr_sw_in'] = df[alt_sw_in]
+    return df
+
+def snow_density_fraction(df, air_t_col):
+    """
+    Estimate density and fraction of snow based on air temp. Translated from IDL code 
+    written by TH Painter, and provided by S Skiles.
+    
+    :param df: pandas df to pull temp col from
+    :param air_t_col: air temp col (in deg C)
+    :return: df with new cols "fraction" and "density"
+    """
+    df['fraction'] = df[air_t_col].copy()
+    df['density'] = df[air_t_col].copy()
+    
+    def transform_fraction(x):
```

Comment:
Personally I avoid nesting functions within functions unless necessary. Usually they can be equally defined at same level and code is easier to read.

Reply:
Good to know. I will avoid nesting in future.